lib/helper/authenticate_helper.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import '../widgets/sign_in_widget.dart';
+import '../widgets/sign_up_widget.dart';
+
+class Authenticate extends StatefulWidget {
+  @override
+  _AuthenticateState createState() => _AuthenticateState();
+}
+
+class _AuthenticateState extends State<Authenticate> {
+  bool showSignIn = true;
+
+  void toggleView() {
+    setState(() {
+      showSignIn = !showSignIn;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (showSignIn) {
+      return SignIn(toggleView);
```

Comment:
if you name this widget's file as Widget, this SignIn should be named as SignInWidget, or SignInScreen which is more describable